### PR TITLE
add scope lesson

### PR DESCRIPTION
**Pull request by esalinas00**
New lesson scope in english

### CODE REVIEW

**Comment by martinheidegger** [comment]
Before translation should start we should review it in english :)

**Comment by esalinas00** [comment]
in english**.

sorry, what do you mean?

Are you going to review my pull request first?, or are you saying that we gonna wait for translation?

**Comment by martinheidegger** [comment]
We should review this PR before someone starts to translate it :smile:

**Comment by esalinas00** [comment]
ok, cool :)

**Comment by SomeoneWeird** [comment]
I'll look through this properly tomorrow, but fwiw under certain conditions variables can be hosted out of functions (not IIFEs). And yes, ideally we should get a "This problem has not.yet been translated" message for each of the.languages we support, so we can still merge stuff like this without breaking anything.

**Comment by esalinas00** [comment]
The concept of global variables is mentioned, and IIFE are mentioned as a way of creating another scope not as the only way.

I didn't explain that JS doesn't have block scope because it will make the lesson considerably longer.

File: problems/scope/problem.md [hunk 1]
```diff
@@ -1,5 +1,70 @@
...
+	var b = 5, c = 6;
+
+	(function secondFunction(){
+		var b = 8;
+		
+		(function thirdFunction(){
+			var a = 7, c = 9;
+
+			(function fourthFunction(){
+				var a = 7, c = 9;
```
Comment (marocchino) [inline problems/scope/problem.md]: We could change values here.
how about `var a = 1, c = 8;`?

File: problems/scope/problem.md [hunk 2]
```diff
@@ -1,5 +1,70 @@
...
+
+			})();
+		})();
+	})();
+})();
+```
+
+Use your knowledge of the variables' `scope` and place the following
+code inside on of the functions in 'scope.js'
```
Comment (marocchino) [inline problems/scope/problem.md]: I want remove newline here.

File: problems/scope/problem.md [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+
+foo(); // 4, 2, 48
+```
+IIFE, Immediately Invoked Function Expression, is a common pattern for creating local scopes
+example:
+```js
+	(function(){ //the function expression is surrounded by parenthesis
+		//variables defined here
+		//can't be accesed outside
+	})() //the function is immediately invoked
```
Comment (marocchino) [inline problems/scope/problem.md]: a semicolon has been omitted.

File: problems/scope/problem.md (hunk 1 above)
Comment (esalinas00) [inline problems/scope/problem.md]: my bad, i didn't noticed I repeated the values.

File: problems/scope/problem.md (hunk 2 above)
Comment (esalinas00) [inline problems/scope/problem.md]: you mean at line 63 right?

File: problems/scope/problem.md (hunk 3 above)
Comment (esalinas00) [inline problems/scope/problem.md]: should I commit and push this changes to my branch?

File: problems/scope/problem.md (hunk 2 above)
Comment (marocchino) [inline problems/scope/problem.md]: umm.. to me, it's line 64 but, the point is "no newline needed middle of sentence."

File: problems/scope/problem.md (hunk 3 above)
Comment (marocchino) [inline problems/scope/problem.md]: yes. :)

File: problems/scope/problem.md (hunk 2 above)
Comment (esalinas00) [inline problems/scope/problem.md]: ok

**Comment by SomeoneWeird** [comment]
JS does have block scope (ES6) but that's outside the scope of this lesson (pardon the pun)

**Comment by esalinas00** [comment]
hehe
There is block scope in try/catch blocks too, but explaining it in the same lesson would be to much.

**Comment by marocchino** [comment]
:+1: for me. :clap:

File: problems/scope/problem.md [hunk 4]
```diff
@@ -1,5 +1,69 @@
...
 
-#
+# SCOPE
 
----
+`Scope` is the set of variables, objects, and functions you have access to.
+
+JavaScript has two scopes: `global` and `local`. A variable that is declared outside a function definition is a `global` variable, and its value is accessible and modifiable throughout your program. A variable that is declared inside a function definition is `local`. It is created and destroyed every time the function is executed, and it cannot be accessed by any code outside the function.
+
+Functions defined inside other functions, kwown as nested functions, have access to their parent function's scope.
```
Comment (SomeoneWeird) [inline problems/scope/problem.md]: typo

**Comment by SomeoneWeird** [comment]
Also it would be great if you could add a space after starting all your comments `//`, just to keep code style similar. I've not tested it, but if it works (after fixing typo and spaces) we can merge it in :+1:

**Comment by esalinas00** [comment]
cool, I fixed the typo and added the spaces.

**Comment by SomeoneWeird** [comment]
Awesome! Is it possible for you to squash you commits? If you don't know how that's cool, we can merge it in anyway.

**Comment by esalinas00** [comment]
I squashed the commits. :smile: 
(It was my first time doing it, outside of guides)

**Comment by SomeoneWeird** [comment]
Awesome! :tada:

**Pull request merged**